lib/util/http/index.ts

```diff
@@ -57,6 +57,40 @@ async function resolveResponse<T>(
   }
 }
 
+// isAmazon return true if request options contains Amazon related headers
+function isAmazon(opts: any): boolean {
+  return opts.search?.includes('X-Amz-Algorithm');
+}
+
+// isAzureBlob return true if request options contains Azure container registry related data
+function isAzureBlob(opts: any): boolean {
+  return (
+    opts.hostname?.endsWith('blob.core.windows.net') &&
```

Comment:
```suggestion
    opts.hostname?.endsWith('.blob.core.windows.net') &&
```
just make sure we also match the dot